src/lib.rs

```diff
@@ -149,12 +148,15 @@ impl Piranha {
       .set_language(*piranha_args.language().language())
       .expect("Could not set the language for the parser.");
 
-    let temp_dir = self.write_code_snippet_to_temp();
+    let mut path_to_codebase = self.piranha_arguments.path_to_codebase().to_string();
 
-    let path_to_codebase = temp_dir
-      .as_ref()
-      .map(|t| t.path().to_str().unwrap_or_default().to_string())
-      .unwrap_or(self.piranha_arguments.path_to_codebase().to_string());
+    let temp_dir = if !self.piranha_arguments.code_snippet().is_empty() {
+      let td = self.write_code_snippet_to_temp();
+      path_to_codebase = td.path().to_str().unwrap_or_default().to_string();
```

Comment:
So, this is still silently overriding `self.piranha_arguments.path_to_codebase()` when both are set, no?

I still think this unintuitive and we should be erroring out when both arguments are present (or else merging their contents, but that seems harder and probably outside the scope of this PR). See comment on `_validate(...)` in `piranha_arguments.rs` below.